api/repositories/shared.go

```diff
@@ -1,15 +1,56 @@
 package repositories
 
 import (
+	"context"
 	"errors"
+	"fmt"
 
 	"code.cloudfoundry.org/cf-k8s-controllers/api/authorization"
 	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
 
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+var CFAppGVR = schema.GroupVersionResource{
+	Group:    "workloads.cloudfoundry.org",
+	Version:  "v1alpha1",
+	Resource: "cfapps",
+}
+
+type NamespaceRetriever struct {
+	client dynamic.Interface
+}
+
+func NewNamespaceRetriver(client dynamic.Interface) NamespaceRetriever {
+	return NamespaceRetriever{
+		client: client,
+	}
+}
+
+func (n NamespaceRetriever) GetNamespacedResourceNamespace(ctx context.Context, resourceGUID string, gvr schema.GroupVersionResource) (string, error) {
```

Comment:
Maybe the name can be GetResourceNamespace? Sounds a bit redundant to me in its current form.

Reply:
Naming is hard. Thanks for the guidance.

Reply:
+1 for better name. How about `NamespaceFor(...)`?